### DISCUSSION TITLE
Search ASOS

### DISCUSSION THREAD

**Discussion by vaskort** [johnlindquist/kit#485] [Share]
It will search at ASOS.com

[Install search-asos](https://scriptkit.com/api/new?name=search-asos&url=https://gist.githubusercontent.com/vaskort/2c7dcba1649783d51d1df2cb860ab2dd/raw/988c94b6eba344a61d9b9a0ee51c7b7e70870a07/search-asos.js)

```js
// Shortcode: asos
// Menu: search-asos
// Description: Search ASOS
// Author: Vasilis Kortsimelidis
// Twitter: @vas_kort

let searchTerm = await arg("ASOS term to search for:")
let termWithEncodedSpaces = searchTerm.replace(/ /g, "%20");

exec(`open https://www.asos.com/search/?q=${termWithEncodedSpaces}`)
```

**Comment by johnlindquist** [comment]
Awesome, love seeing scripts put together for simple time-savers.

Quick note, you don't need the surrounding `Search query:` arg, this should be enough:

```js
let searchTerm = await arg("ASOS term to search for:")

exec(`open https://www.asos.com/search/?q=${searchTerm}`)
```

**Reply by vaskort** [reply]
Ah thank you I copied the logic from another script! I will update. 

Also I realised (after opening this thread) that the above fails to execute when you search a term that includes two words, for instance `nike air` do you have any quick pointers around that?

**Reply by johnlindquist** [reply]
I think you'll want to surround `searchTerm` in quotes when passing it to exec:

before
```js
exec(`open https://www.asos.com/search/?q=${searchTerm}`)
```

after (with quotes):
```js
exec(`open https://www.asos.com/search/?q="${searchTerm}"`)
```

You have to think of how the terminal would handle this command:

`open https://www.asos.com/search?q=Nike Air`

`Air` would be considered the 2nd argument passed to `open`

**Reply by vaskort** [reply]
ah! thanks for the explanation that makes sense!

I tried debugging in the terminal but I couldn't make it work. I think in case of passing url `open` expects a string without spaces. 
I thought just encoding the spaces should be enough for this!

Updated code below: 

```
let termWithEncodedSpaces = searchTerm.replace(/ /g, "%20");

exec(`open https://www.asos.com/search/?q=${termWithEncodedSpaces}`)
```

**Reply by johnlindquist** [reply]
Another option is to wrap the entire url in quotes:

`open "https://scriptkit.com"`
